### DISCUSSION TITLE
Problem building libmesh

### DISCUSSION THREAD

**Discussion by hsheldon** [idaholab/moose#18929] [Q&A Getting Started] [answered]
I'm having trouble building libmesh. Here's what I'm getting:

libtool: warning: '/apps/gcc/9.2.0/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../lib64/libstdc++.la' seems to be moved
libtool: warning: '/apps/gcc/4.9.3/lib/../lib64/libgfortran.la' seems to be moved
libtool: warning: '/apps/gcc/4.9.3/lib/../lib64/libquadmath.la' seems to be moved
libtool: warning: '/apps/gcc/4.9.3/lib/../lib64/libquadmath.la' seems to be moved
libtool: warning: '/apps/gcc/4.9.3/lib/../lib64/libgfortran.la' seems to be moved
libtool: warning: '/apps/gcc/4.9.3/lib/../lib64/libquadmath.la' seems to be moved
libtool: warning: '/apps/gcc/4.9.3/lib/../lib64/libquadmath.la' seems to be moved
libtool: warning: '/apps/gcc/4.9.3/lib/../lib64/libgfortran.la' seems to be moved
libtool: warning: '/apps/gcc/4.9.3/lib/../lib64/libquadmath.la' seems to be moved
libtool: warning: '/apps/gcc/4.9.3/lib/../lib64/libquadmath.la' seems to be moved
libtool: warning: '/apps/gcc/4.9.3/lib/../lib64/libquadmath.la' seems to be moved
libtool: warning: '/apps/gcc/4.9.3/lib/../lib64/libquadmath.la' seems to be moved
  CXXLD    fparser_parse-opt
./.libs/libmesh_opt.so: error: undefined reference to 'GOMP_loop_nonmonotonic_dynamic_start'
./.libs/libmesh_opt.so: error: undefined reference to 'GOMP_loop_nonmonotonic_dynamic_next'
collect2: error: ld returned 1 exit status
Makefile:13596: recipe for target 'fparser_parse-opt' failed
make[1]: *** [fparser_parse-opt] Error 1
make[1]: Leaving directory '/scratch1/projects/moose/moose/libmesh/build'
Makefile:32290: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1

In those warnings it seems to be looking for gcc/4.9.3 but I have gcc/9.2.0 loaded. Does that have something to do with it? Or is the error completely unrelated to the warnings?
Cheers,
Heather

**Comment by GiudGiud** [comment, marked as answer]
Hello

Seems like remnants of a previous compilation are being used. Could you please clean the repo (save your work then git clean -xfd) then try again.

How are you building libmesh? With scripts/update_and_rebuild_libmesh?

Guillaume

**Reply by hsheldon** [reply]
I've started over with a completely clean installation - just using plain vanilla MOOSE, not my own MOOSE app - still the same problem. I'm using scripts/update_and_rebuild_libmesh. Presumably I need to figure out why it's looking for gcc/4.9.3?

Thanks,
Heather

**Reply by loganharbour** [reply]
Heather - I suspect gcc/4.9.3 is not an issue here. Compilers are needed to compile... compilers. In this case, I suspect your server admins use gcc 4.9.3 to compile gcc 9.2.0.

**Reply by hsheldon** [reply]
Ok, I'll stop going down the 4.9.3 rabbit hole! Any other suggestions?

**Reply by GiudGiud** [reply]
what does `which mpicxx` return though?

**Reply by hsheldon** [reply]
/apps/openmpi/3.1.4-ofed45-gcc/bin/mpicxx

**Reply by milljm** [reply]
Can you list all your loaded modules? And a printout of an `env` would also be helpful:

```bash
module purge
module load <the modules you are attempting to use>
module list
env
```

Thanks!

**Reply by hsheldon** [reply]
I seem to have fixed the problem by switching from gcc/9.2.0 to gcc/6.1.0. I have the following modules loaded:

Currently Loaded Modulefiles:
  1) SC                              4) openmpi/3.1.4-ofed45-gcc        7) netcdf/4.3.3.1                 10) hdf5/1.8.16-gcc                13) git/2.30.0                     16) paraview/5.5.0
  2) slurm/current                   5) petsc/3.10.5-gcc-mpi314-moose   8) glpk/4.60                      11) eigen/3.3.7                    14) python/2.7.13
  3) hpc-x/2.4.0                     6) intel-mkl/11.3.2                9) gcc/6.1.0                      12) moose/master                   15) mesa/17.2.4_gallium


Maybe there's something in there that explains why gcc/6.1.0 works?

**Reply by GiudGiud** [reply]
As long as all of these were compiled with gcc 6.1.0 (or 9.2 previously)
Why are you loading moose/master though? You are compiling your own libmesh, you should compile your own moose too

**Reply by hsheldon** [reply]
I am compiling moose as well - which works now that I've managed to build libmesh. I have module load moose in my .profile.
Thanks for your help!
